tests/testthat/test-target_sda.R

```diff
@@ -208,23 +170,91 @@ test_that("outputs 3 metrics of CO2 emissions", {
     )
   )
 
-  expect_length(unique(out$emission_factor_metric), 3L)
+  exp <- c("sector", "year", "emission_factor_metric", "emission_factor_value")
+  expect_named(out, exp)
 })
 
-test_that("outputs expected names", {
-  out <- target_sda(
-    fake_matched(sector_ald = "cement"),
-    ald = fake_ald(
-      sector = "cement",
-      technology = "cement",
-      year = c(2020, 2021, 2022),
-      emission_factor = c(1, 2, 3)
-    ),
-    co2_intensity_scenario = fake_co2_scenario(
-      year = c(2020, 2050), emission_factor = c(0.6, 0.2)
+test_that("with known input outputs as expected", {
+  #TODO: Re-factor this test into smaller isolated expected output tests
+matched <- fake_matched(sector_ald = "cement")
+
+ald <- fake_ald(
+        sector = "cement",
+        technology = "cement",
+        name_company = c(rep("shaanxi auto", 4), "company 2"),
+        year = c(2020, 2021, 2022, 2025, 2020),
+        emission_factor = c(0.9, 0.9, 0.8, 0.5, 12)
+      )
+
+
+co2_intensity_scenario <- fake_co2_scenario(
+  scenario = c(rep("b2ds", 2), rep("sds", 2)),
+  year = rep(c(2020, 2025), 2),
+  emission_factor = c(0.5, 0.1, 0.5, 0.4)
+
+)
+
+out <- target_sda(matched,
+                  ald,
+                  co2_intensity_scenario) %>%
+  arrange(.data$year)
+
+out <- split(out, out$emission_factor_metric)
+
+expected_out <- tribble(
+   ~sector, ~year,  ~emission_factor_metric, ~emission_factor_value,
+  "cement", 2020L,              "projected",                    0.9,
+  "cement", 2021L,              "projected",                    0.9,
+  "cement", 2022L,              "projected",                    0.8,
+  "cement", 2025L,              "projected",                    0.5,
+  "cement", 2020L,            "target_b2ds",                    0.9,
+  "cement", 2025L,            "target_b2ds",                   1.29,
+  "cement", 2020L,             "target_sds",                    0.9,
+  "cement", 2025L,             "target_sds",                   5.16,
+  "cement", 2020L, "adjusted_scenario_b2ds",                   6.45,
+  "cement", 2025L, "adjusted_scenario_b2ds",                   1.29,
+  "cement", 2020L,  "adjusted_scenario_sds",                   6.45,
+  "cement", 2025L,  "adjusted_scenario_sds",                   5.16,
+  "cement", 2020L,      "corporate_economy",                   6.45,
+  "cement", 2021L,      "corporate_economy",                    0.9,
+  "cement", 2022L,      "corporate_economy",                    0.8,
+  "cement", 2025L,      "corporate_economy",                    0.5
+  )
```

Comment:
Is this dead code? `expected_out` seems unused.

Reply:
Yup you are correct! Thanks